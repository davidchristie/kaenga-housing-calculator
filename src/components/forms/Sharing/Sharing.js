import React, { Component, PropTypes } from 'react'
import { ControlLabel, FormGroup, Panel } from 'react-bootstrap'
import { Field } from 'redux-form'

import Checkbox from '../../controls/Checkbox'

class Sharing extends Component {
  render () {
    const { handleSubmit } = this.props
    return (
      <form onSubmit={handleSubmit}>
        <Panel header='Sharing'>
          <FormGroup>
            <ControlLabel>Are you willing to share:</ControlLabel>
            <Field
              component={Checkbox}
              label='Laundries and outdoor areas'
              name='laundries'
            />
            <Field
              component={Checkbox}
              label='Guest rooms'
              name='guestRooms'
            />
            <Field
              component={Checkbox}
              label='Entertainment spaces'
              name='entertainment'
            />
          </FormGroup>
        </Panel>
      </form>
    )
  }
}

Sharing.propTypes = {
  handleSubmit: PropTypes.func
}

export default Sharing
