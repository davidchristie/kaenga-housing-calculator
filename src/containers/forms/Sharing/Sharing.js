import { reduxForm } from 'redux-form'

import Sharing from '../../../components/forms/Sharing'

export default reduxForm({
  form: 'sharing'
})(Sharing)
