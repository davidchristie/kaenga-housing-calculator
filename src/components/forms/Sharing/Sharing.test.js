import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import Sharing from './Sharing'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<Sharing />)).toMatchSnapshot()
})
