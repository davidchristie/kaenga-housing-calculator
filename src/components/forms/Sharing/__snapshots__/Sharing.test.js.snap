exports[`test matches snapshot 1`] = `
<form
  onSubmit={undefined}>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}
    header="Sharing">
    <FormGroup
      bsClass="form-group">
      <ControlLabel
        bsClass="control-label"
        srOnly={false}>
        Are you willing to share:
      </ControlLabel>
      <Field
        component={[Function]}
        label="Laundries and outdoor areas"
        name="laundries" />
      <Field
        component={[Function]}
        label="Guest rooms"
        name="guestRooms" />
      <Field
        component={[Function]}
        label="Entertainment spaces"
        name="entertainment" />
    </FormGroup>
  </Panel>
</form>
`;
